src/foam/box/network/SocketServer.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.box.network',
+  name: 'SocketServer',
+
+  implements: [
+    'foam.nanos.NanoService'
+  ],
+
+  javaImports: [
+    'foam.box.Box',
+    'foam.core.X',
+    'foam.core.ContextAware',
+    'foam.nanos.logger.Logger',
+    'java.net.ServerSocket',
+    'java.net.Socket',
+    'java.io.IOException'
+  ],
+
+  properties: [
+    {
+      class: 'Int',
+      name: 'port',
+      value: 7070
+    }
+  ],
+
+  methods: [
+    {
+      name: 'start',
+      javaCode: `
+        try {
+          System.out.println("Starting TCP Server on port: " + getPort());
+          ServerSocket serverSocket = new ServerSocket(getPort());
+          (new Thread() {
```

Comment:
Using ThreadPoolExecutor will provide thread pool support along with control over Daemon and Priority. 
See foam/nanos/pool/ThreadPoolAgency